_posts/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins.md

```diff
@@ -0,0 +1,336 @@
+
+<img src="/assets/media/blog-images/2024-05-01-develop-guideline-multiple-data-source-in-opensearch-and-plugins/multi_selectable.png">
+
+### Data sources as saved objects
+
+The data source is persisted as a saved object in the OpenSearch Dashboards system index. Upon creation, it is assigned a UUID as a unique identifier, and metadata such as the remote cluster version and associated plugins is gathered. This automatically generated data, along with user-provided information like the title, endpoint, and authentication methods, is stored as attributes of the data source object. Refer to the [related code](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data_source/common/data_sources/types.ts#L8) for the current object schema. 
+
+
+## List of supported plugins/features
+
+Starting from version 2.14, the following external OpenSearch Dashboards plugins support multiple data sources: 
+
+* Index Management
+* Machine Learning
+* Search Relevance
+* Anomaly Detection
+* Maps
+* Security
+* Notification
+* Query Workbench
+* Trace Analytics 
+
+Multiple data sources functionality is also now available with Time Series Visual Builder (TSVB), Vega and Region Map visualization types.
+
+## Frequently asked questions 
+
+Q: Is any UI component provided by `dataSourceManagemet` used in core OpenSearch?
+A: A data source selector is now used in Dev Tools, the tutorial, and on the saved object management page. You can see the component after setting `data_source.enabled` to `true` in your `opensearch_dashbards.yml` file.
+
+Q: Since a dashboard is a saved object, won't it have an associated data connection?
+A: A data source, represented as a saved object, is identified by its unique saved object ID. Saved objects created from data sources have a `references` section containing the ID, name, and type. This reference helps identify the associated data source for the saved object. Here's an example of an index pattern reference:
+```
+"references": [
+    {
+        "id": "9dc4e190-08a0-11ef-901b-bb0301313422",
+        "type": "data-source",
+        "name": "dataSource"
+    }
+]
+```
+The visualization reference is slightly different as it points to the related index pattern and uses the index pattern ID to find the corresponding data source, as shown in the following example:
+```
+"references": [
+    {
+        "name": "kibanaSavedObjectMeta.searchSourceJSON.index",
+        "type": "index-pattern",
+        "id": "f1613e10-08a6-11ef-901b-bb0301313422"
+        }
+]
+```
+
+## Conclusion
+
+This blog post covered the process of integrating OpenSearch Dashboards plugins with multiple data sources. We encourage you to experiment with multiple data sources using the provided example plugins, developed by Lu Yu. This will allow you to experience the benefits of accessing multiple data sources within a single dashboard and enable your plugin to expose a wider range of data from remote clusters.
```

Comment:
Should we include Lu Yu's GitHub handle?

Reply:
I have provided how to install example plugins above